bin/test/main.ml

```diff
@@ -258,6 +260,8 @@ let update_file_or_block ?root ppf md_file ml_file block direction =
   | `To_ml ->
      update_file_with_block ppf block ml_file (Block.part block)
 
+exception Test_block_failure of Block.t * string
```

Comment:
Again, I think I'd prefer if we used results instead of exceptions there but given this is already an improvement we can refactor that later.